Minio/DataModel/ObjectOperationsArgs.cs

```diff
@@ -0,0 +1,134 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using RestSharp;
+using Minio.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.Xml.Linq;
+
+namespace Minio
+{
+    public class RemoveObjectArgs : ObjectArgs<RemoveObjectArgs>
+    {
+        internal string VersionId { get; private set; }
+        internal bool AllVersions { get; private set; }
+
+        public RemoveObjectArgs()
+        {
+            this.RequestMethod = Method.DELETE;
+        }
+
+        public override void Validate()
+        {
+            base.Validate();
+            if (!string.IsNullOrEmpty(this.VersionId) && this.AllVersions)
+            {
+                throw new MinioException("AllVersions flag cannot be set along with the VersionId");
+            }
+        }
+        public override RestRequest BuildRequest(RestRequest request)
+        {
+            if (!string.IsNullOrEmpty(this.VersionId))
+            {
+                request.AddQueryParameter("versionId",$"{this.VersionId}");
+            }
+            return request;
+        }
+        public RemoveObjectArgs WithVersionId(string ver)
+        {
+            this.VersionId = ver;
+            return this;
+        }
+        public RemoveObjectArgs WithRemoveAllVersions(bool rem)
+        {
+            this.AllVersions = rem;
+            return this;
+        }
+    }
+
+    public class RemoveMultipleObjectArgs : ObjectArgs<RemoveMultipleObjectArgs>
+    {
+        internal List<string> ObjectNames { get; set; }
+        internal List<Tuple<string, List<string>>> ObjectNamesVersions  { get; set; }
+        internal bool AllVersions { get; set; }
+
+        public RemoveMultipleObjectArgs()
+        {
+            this.ObjectNames = new List<string>();
+            this.RequestMethod = Method.POST;
+        }
+
+        internal RemoveMultipleObjectArgs(RemoveObjectArgs rmArgs, List<string> versionIDs)
+        {
+            this.ObjectNames = new List<string>();
+            this.ObjectNamesVersions = new List<Tuple<string, List<string>>>();
+            this.RequestMethod = Method.POST;
+            this.BucketName = rmArgs.BucketName;
+            this.ObjectName = rmArgs.ObjectName;
+            if (rmArgs.AllVersions)
+            {
+                if (versionIDs == null || versionIDs.Count < 1)
+                {
+                    throw new MinioException("AllVersions object set but List of version IDs is not set.");
+                }
+                this.AllVersions = true;
+                WithAddObjectNameWithVersions(rmArgs.ObjectName, versionIDs);
+            }
+        }
+
+        public RemoveMultipleObjectArgs WithAddObjectName(string name)
```

Comment:
```suggestion
        public RemoveMultipleObjectArgs WithObjectName(string name)
```

Reply:
WithBucketName/WithObjectName are already inherited from BucketArgs/ObjectArgs. I'll modify the name.

Reply:
Changed to `WithSingleObject`